bake/bake.go

```diff
@@ -1179,6 +1179,16 @@ func updateContext(t *build.Inputs, inp *Input) {
 	t.ContextState = &st
 }
 
+func isRemoteContext(t build.Inputs, inp *Input) bool {
+	if build.IsRemoteURL(t.ContextPath) {
+		return true
+	}
+	if inp != nil && build.IsRemoteURL(inp.URL) && !strings.HasPrefix(t.ContextPath, "cwd://") {
```

Comment:
isn't `inp.URL` always URL if it is set?

Reply:
Yes if a remote bake def is set but if context is set to git context it is not (first cond in `isRemoteContext`):

```
docker buildx bake --set *.context=https://github.com/docker/buildx.git --print
{
  "group": {
    "default": {
      "targets": [
        "binaries"
      ]
    }
  },
  "target": {
    "binaries": {
      "context": "https://github.com/docker/buildx.git",
      "dockerfile": "Dockerfile",
      "args": {
        "BUILDKIT_CONTEXT_KEEP_GIT_DIR": "1"
      },
      "target": "binaries",
      "platforms": [
        "local"
      ],
      "output": [
        {
          "dest": "./bin/build",
          "type": "local"
        }
      ]
    }
  }
}
```

Reply:
Isn't that the `build.IsRemoteURL(t.ContextPath) {`, how does that affect `inp.URL`?

Reply:
> Isn't that the `build.IsRemoteURL(t.ContextPath) {`,

Yes it is but `inp.URL` is only set when remote bake def arg is set: https://github.com/docker/buildx/blob/026ac2313c75cb374a25b22474e1ba364a3fa323/commands/bake.go#L557

It doesn't take into account git urls in context within bake def or override.